wolf-data/src/Wolf/Data/Export/Types.hs

```diff
@@ -89,3 +143,65 @@ instance ToJSON Repo where
             , "notes" .= repoNotes
             , "suggestions" .= repoSuggestions
             ]
+
+type Warn = [Problem]
+
+data Problem
+    = WarnMissingNoteIndex PersonUuid
+                           NoteUuid
+    | WarnMissingNote NoteUuid
+    | WarnMissingRelevantPerson PersonUuid
+                                NoteUuid
+    | WarnMissingRelevantNote NoteUuid
+                              PersonUuid
+    deriving (Show, Eq, Generic)
+
+instance ToJSON Problem
+
+instance FromJSON Problem
+
+data Err =
+    NoInitFile
+    deriving (Show, Eq, Generic)
+
+instance ToJSON Err
+
+instance FromJSON Err
+
+type CautiousS = CautiousT Warn Err
+
+cautiousProblem :: Monad m => Problem -> a -> CautiousS m a
+cautiousProblem p = cautiousWarning [p]
+
+cautiousProblemIfNothing ::
+       Monad m => Problem -> Maybe a -> CautiousS m (Maybe a)
+cautiousProblemIfNothing p = cautiousWarningIfNothing [p]
```

Comment:
It seems all of these should be in the cautious library instead of here.

Reply:
The cautious library has no "problem".